Dockerfile

```diff
@@ -14,17 +14,21 @@ ENV PYTHON_VERSION_36 3.6.12
 ENV PYTHON_VERSION_37 3.7.9
 ENV PYTHON_VERSION_38 3.8.6
 ENV PYTHON_VERSION_39 3.9.1
+ENV PYTHON_VERSION_310 3.10.0b3
 ENV PYPY_VERSION_35 pypy3.5-7.0.0
-# Note: 4.7.12.1 drastically increases memory usage
-ENV CONDA_VERSION 4.6.14
+# Latest miniconda version is Miniconda3-py39_4.10.1, but it's not available for x86_64
+ENV CONDA_PYTHON_VERSION py38
+ENV CONDA_VERSION 4.9.2
```

Comment:
I added `CONDA_PYTHON_VERSION` here as well, because they have a Python version in these new releases as well. However, they also have the "generic" one `Miniconda3`: `Miniconda3-4.4.10-Linux-x86_64.sh`

That version is newer/bigger (4.4.10) than the one being installed here. So, maybe it's better to install the generic Python3 version but the latest conda version. cc @astrojuanlu

Reply:
I'm not following, why wouldn't we want to upgrade now to conda 4.4.10? Not sure if we need to pin the Python version of the base environment

Reply:
I think we always want to upgrade to the latest miniconda available unless we know there is an issue with it (as already happened some time ago).

However, the official docs https://docs.conda.io/en/latest/miniconda.html#linux-installers points to a version https://repo.anaconda.com/miniconda/Miniconda3-py39_4.9.2-Linux-x86_64.sh that's not the "latest" in the directory index (https://repo.anaconda.com/miniconda/) where the latest that I found there is 4.4.10 without a Py3 version pinned.

Besides, the SHA256 from the official docs (`536817d1b14cb1ada88900f5be51ce0a5e042bae178b5550e62f61e223deae7c`) is not listed in the index directory neither. So, 🤷🏼 

This is why I'm confused and I'm not sure what's the correct version to upgrade to.

Reply:
I think there's a temporary issue with the Miniconda installers and we should use 4.10 rather than 4.9 even if the `latest` pointer is outdated, WDYT @jezdez?

Reply:
I got confused between `4.10.1` and `4.4.10` (because I suppose that I searched for the `4.10` part only). `4.10.1` is not built for `x86_64`, so we can't use that one. The one that I originally picked is the latest version we can use: Py3.9 Miniconda 4.9.2